src/docs/ad/ad-service-discovery.rst

```diff
@@ -0,0 +1,87 @@
+Domain
+    A domain is defined as a logical group of network objects (computers, users, devices) that share the same Active Directory database.
+
+Forest
+    A forest is a collection of trees that share a common global catalog, directory schema, logical structure, and directory configuration. A forest contains one or multiple domains.
+
+Site
+    A logical organization, or container for objects within a domain. `Sites <https://docs.microsoft.com/en-us/windows-server/identity/ad-ds/plan/understanding-active-directory-site-topology>`_ are often used to group geographically-close systems to together. Clients can prefer to communicate only to servers within a site.
+
+Global Catalog
+    A service advertising information about Active Directory domains in a forest, it can be searched by clients similarly to normal LDAP user/group information. The `Global catalog <https://docs.microsoft.com/en-us/windows/win32/ad/global-catalog>`_ is used by SSSD clients to read and discovery AD Site and Forest information, and trusted domains.
+
+CLDAP Ping
+    See section below
+
+When using the AD provider, by default SSSD clients utilize auto-discovery of Active Directory servers to help avoid explicitly configuring Active Directory servers, and enable failover behavior. The term **Service Discovery** may also be used in SSSD documentation and manual pages to describe this.
+
+
+LDAP/CLDAP Ping
+===============
+`C(Connection-Less)LDAP Ping <https://ldapwiki.com/wiki/LDAP%20ping>`_ is a way for clients to check the liveliness of a domain controller. It is often necessary for SSSD to try to find available domain controllers within a domain, or within an AD Site. SSSD sends a LDAP Ping during the service discovery phase by sending a `rootDSE <https://ldapwiki.com/wiki/RootDSE>`_ search of the **netlogon** attribute. If available, the AD domain controller will return site, forest, and domain information in the **netlogon** response.
+
+.. note::
+
+    C(Connection-Less)LDAP Ping behavior performed by Microsoft clients is described in `Microsoft protocol documentation <https://winprotocoldoc.blob.core.windows.net/productionwindowsarchives/WinArchive/%5bMS-DISO%5d.pdf>`_
+
+    .. note::
+
+    SSSD tries to reuse previously stored AD site and forest information, CLDAP pings are only performed if SSSD is starting up, or recovering from an offline state
+
+Site-enabled discovery
+======================
+At domain initialization, if ``ad_enable_dns_sites`` is true (default) then the 'AD' SRV lookup plugin is set. SSSD will attempt to discover primary(site-specific) servers and backup servers using the following steps:
+
+.. note::
+
+    The **ad_site** option can be set manually in the sssd configuration to speed up the discovery process.
+
+#. **Set discovery domain**: read from ``dns_discovery_domain`` option or use ``ad_domain`` name
+#. **Lookup AD site and forest** information using CLDAP Ping
+
+.. image:: cldap-ping.svg
+   :height: 700px
+   :align: center
```

Comment:
Nice image! But it is distorted, you need to set smaller height.

Make the Batch 3 arrow connect with batch 2 and 1 arrows?